### ISSUE TITLE
BackgroundGeolocation: where to put my custom script?

### ISSUE THREAD

**Issue by djereg** [driftyco/ionic-native#1196]
Since 2.6.0 the `BackgroundGeolocation.configure()` function has changed, and accepts only the configuration object. But now where can I put my custom script, which I was used before.

I have to stop the background tracking when the user arrived the "destination". But how?

**Comment by djereg** [comment]
Just a simple is inside the polygon check.

```typescript
(location) => {
  this.zone.run(() => {
    if (this.inAreaOf(location.latitude, location.longitude, this.place)) {
      BackgroundGeolocation.stop();
    }
  }
}
```

The user manually starts the tracking, and the plugin sends the user's location to server, until he arrives the specific destination.

**Comment by ihadeed** [comment]
Oh ok. It returns back a promise now.

I should change that and return an observable instead, since the callback will be called multiple times.

**Comment by djereg** [comment]
Yes it returns a promise and it correctly returns the location, the first, only once.

Thank you.

**Comment by ihadeed** [comment]
Fixed by https://github.com/driftyco/ionic-native/commit/961cff1 and published as `2.9.0`
